### DISCUSSION TITLE
Drozone returning array_filter() error

### DISCUSSION THREAD

**Discussion by logifloor** [Laravel-Backpack/community-forum#772] [Q&A (Help)]
I'm struggling with Dropzone uploads when using the Spatie media library addon.  All is working well until I try and submit a form without any uploads (other fields text/date etc populated).  I don't want to 'require' an upload because sometimes not necessary for this application, however, after extensive debugging, I can't find a solution to array_filter(): Argument #1 ($array) must be of type array, null given, when the files field is submitted empty.  Anybody else come across this?

**Comment by logifloor** [comment]
Well, theres 5 hours of my life I'll never get back.

The problem seems to revolve around this code in medialibrary-uploaders/ MediaAjaxUploader.php

```
$uploads = json_decode($uploads, true) ?? [];

$uploadedFiles = array_filter($uploads, function ($value) use ($temporaryFolder, $temporaryDisk) {
    return strpos($value, $temporaryFolder) !== false && Storage::disk($temporaryDisk)->exists($value);
});
```

Its supposed to decode a/the JSON string into an array. If $uploads is null or if json_decode fails and returns null, its supposed to provide an empty array as a default value.  However, It seems that despite the null coalescing operator, $uploads is still null when it reaches array_filter, which is why the error occurs.

I've modified the method in MediaAjaxUploader.php to include an additional check and make sure an empty array is returned even when all else fails.  

```
public function uploadFiles(Model $entry, $value = null)
{
    $temporaryDisk = CRUD::get('dropzone.temporary_disk');
    $temporaryFolder = CRUD::get('dropzone.temporary_folder');

    $uploads = $value ?? CRUD::getRequest()->input($this->getName());

    // Ensure $uploads is an array. If not, decode it or set it to an empty array.
    if (!is_array($uploads)) {
        $uploads = is_string($uploads) ? json_decode($uploads, true) : [];
    }

    // Ensure $uploads is an array before using array_filter
    if (!is_array($uploads)) {
        $uploads = [];
    }

    $uploadedFiles = array_filter($uploads, function ($value) use ($temporaryFolder, $temporaryDisk) {
        return strpos($value, $temporaryFolder) !== false && Storage::disk($temporaryDisk)->exists($value);
    });

    $previousSentFiles = array_filter($uploads, function ($value) use ($temporaryFolder) {
        return strpos($value, $temporaryFolder) === false;
    });

    $previousFiles = $this->get($entry);

    foreach ($previousFiles as $previousFile) {
        if (! in_array($this->getMediaIdentifier($previousFile, $entry), $previousSentFiles)) {
            $previousFile->delete();
        }
    }

    foreach ($uploadedFiles as $key => $value) {
        $file = new File(Storage::disk($temporaryDisk)->path($value));

        $this->addMediaFile($entry, $file);
    }
}
```

**Comment by karandatwani92** [comment]
Hey @logifloor,

Sorry, you had to go through with this! I understand how this has been, spending time on this bug.

I have assigned my colleague to go through with the problem & your proposed solution.

We'll try to get it fixed soon. Thanks for everything!

To be continued here  https://github.com/Laravel-Backpack/medialibrary-uploaders/issues/22
